File: exercises/RB120_OOP/medium_1/word_to_digit.rb
NUM = {
  'one'   => '1',
  'two'   => '2',
  'three' => '3',
  'four'  => '4',
  'five'  => '5',
  'six'   => '6',
  'seven' => '7',
  'eight' => '8',
  'nine'  => '9'
}

def word_to_digit(string)
  string.split.map do |word|
    if NUM.include?(word.downcase)
      NUM[word.downcase]
    elsif NUM.include?(word[0..-2].downcase)
      punc = word.chars.pop
      NUM[word[0..-2].downcase] + punc
    else
      word
    end
  end.join(' ')
end

p word_to_digit('Please call me at five five five one two three four. Thanks.') == 'Please call me at 5 5 5 1 2 3 4. Thanks.'

File: exercises/JS210_Small_Problems/Medium_1/wordToDigit.js
var numStrings = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
var regex = new RegExp(`(${numStrings.join('|')})`, 'ig')
// same as: /zero|one|two|three|four|five|six|seven|eight|nine/ig

function wordToDigit(str) {
  return str.replace(regex, (match => numStrings.indexOf(match)));
}

/*

Replace:
  - for every match String.prototype.replace() finds, it calls the anonymous function
    argument provided to replace().

  - In this case, the for every word digit found, the function replaces the match
    with its index value from the array.

  - Note: the anonymous function is called for every match found by replace().
  
*/

console.log(wordToDigit('Please call me at five five five one two three four. Thanks.'));
// "Please call me at 5 5 5 1 2 3 4. Thanks."